libbeat/common/fleetmode/fleet_mode.go

```diff
@@ -0,0 +1,53 @@
+// Licensed to Elasticsearch B.V. under one or more contributor
+// license agreements. See the NOTICE file distributed with
+// this work for additional information regarding copyright
+// ownership. Elasticsearch B.V. licenses this file to you under
+// the Apache License, Version 2.0 (the "License"); you may
+// not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing,
+// software distributed under the License is distributed on an
+// "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+// KIND, either express or implied.  See the License for the
+// specific language governing permissions and limitations
+// under the License.
+
+package fleetmode
+
+import (
+	"flag"
+
+	"github.com/elastic/beats/v7/libbeat/common"
+)
+
+// Enabled checks to see if filebeat/metricbeat is running under Agent
+// The management setting is stored in the main Beat runtime object, but we can't see that from a module
+// So instead we check the CLI flags, since Agent starts filebeat/metricbeat with "-E", "management.mode=x-pack-fleet", "-E", "management.enabled=true"
+func Enabled() bool {
+	type management struct {
+		Mode    string `config:"management.mode"`
+		Enabled bool   `config:"management.enabled"`
+	}
+	var managementSettings management
+
+	cfgFlag := flag.Lookup("E")
```

Comment:
Not sure if using `-E` for this is the most appropiate way to add this option. I would expect settings passed with `-E` to be supported also in the main configuration file. Not sure if implementing it this way covers both cases.

If this option is expected to be configured only with flags and not with configuration files, is there some reason to don't add it as an explicit flag, like `--management.mode`?

Reply:
Ok, I see this code is not new, so maybe this discussion can be moved to a future issue.

Reply:
I assume that there were some reasons to implement the workaround this way, but I wasn't part of that process. I'm ok to come back to this later.